File: QolmsYappli/QolmsYappli/Models/PartialViewModel/HealthAgeAdviceAreaPartialViewModel.vb
Imports System.Collections.ObjectModel

<Serializable()>
Public NotInheritable Class HealthAgeAdviceAreaPartialViewModel
    Inherits QyPartialViewModelBase(Of HealthAgeViewModel)

#Region "Variable"

    Private _hasData As Boolean = False

    Private _adviceData As New List(Of HealthAgeAdviceItem)()

#End Region

#Region "Public Property"

    Public ReadOnly Property HasData As Boolean

        Get
            Return Me._hasData
        End Get

    End Property

    Public ReadOnly Property AdviceData As ReadOnlyCollection(Of HealthAgeAdviceItem)

        Get
            Return Me._adviceData.ToList().AsReadOnly()
        End Get

    End Property

#End Region

#Region "Private Method"

    Private Sub InitializeBy(items As IEnumerable(Of HealthAgeAdviceItem))

        Me._hasData = False
        Me._adviceData = New List(Of HealthAgeAdviceItem)()

        If items IsNot Nothing AndAlso items.Any() Then
            Me._hasData = True
            Me._adviceData = items.ToList()
        End If

    End Sub

#End Region

#Region "Constructor"

    ''' <summary>
    ''' <see cref="HealthAgeAdviceAreaPartialViewModel" /> クラスの新しいインスタンスを初期化します。
    ''' </summary>
    ''' <remarks></remarks>
    Protected Sub New()

        MyBase.New()

    End Sub

    ''' <summary>
    ''' <see cref="HealthAgeAdviceAreaPartialViewModel" /> クラスの新しいインスタンスを初期化します。
    ''' </summary>
    ''' <remarks></remarks>
    Public Sub New(model As HealthAgeViewModel)

        MyBase.New(model)

        Me.InitializeBy(model.HealthAgeAdviceN)

    End Sub

#End Region

End Class

File: QolmsJotoWebView/Models/PartialViewModel/HealthAgeAdviceAreaPartialViewModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;

namespace MGF.QOLMS.QolmsJotoWebView
{
    [Serializable]
    public sealed class HealthAgeAdviceAreaPartialViewModel : QjPartialViewModelBase<HealthAgeViewModel>
    {
        #region Variable

        private bool _hasData = false;

        private List<HealthAgeAdviceItem> _adviceData = new List<HealthAgeAdviceItem>();

        #endregion

        #region Public Property

        public bool HasData
        {
            get { return this._hasData; }
        }

        public ReadOnlyCollection<HealthAgeAdviceItem> AdviceData
        {
            get { return this._adviceData.ToList().AsReadOnly(); }
        }

        #endregion

        #region Private Method

        private void InitializeBy(IEnumerable<HealthAgeAdviceItem> items)
        {
            this._hasData = false;
            this._adviceData = new List<HealthAgeAdviceItem>();

            if (items != null && items.Any())
            {
                this._hasData = true;
                this._adviceData = items.ToList();
            }
        }

        #endregion

        #region Constructor

        /// <summary>
        /// <see cref="HealthAgeAdviceAreaPartialViewModel" /> クラスの新しいインスタンスを初期化します。
        /// </summary>
        protected HealthAgeAdviceAreaPartialViewModel()
            : base()
        {
        }

        /// <summary>
        /// <see cref="HealthAgeAdviceAreaPartialViewModel" /> クラスの新しいインスタンスを初期化します。
        /// </summary>
        public HealthAgeAdviceAreaPartialViewModel(HealthAgeViewModel model)
            : base(model)
        {
            this.InitializeBy(model.HealthAgeAdviceN);
        }

        #endregion
    }
}
